.github/workflows/nix.yml

```diff
@@ -26,3 +23,41 @@ jobs:
 
       - name: Build all flake outputs
         run: om ci
+
+      - name: Build and push Docker image
+        if: github.event_name == 'push' && github.event.pull_request.merged == true
```

Comment:
Is there a particular reason for this condition?

You can also check on the branch name, right?

See https://github.com/juspay/omnix/blob/1f62a25f92c84ce88029783ad5a1bad1245edec6/.github/workflows/ci.yaml#L11